crates/nu_plugin_start/src/start.rs

```diff
@@ -0,0 +1,178 @@
+            self.filenames.push(filename);
+        } else {
+            print_warning(format!(
+                "The file '{}' does not exist",
+                Color::White.bold().paint(filename)
+            ));
+        }
+    }
+
+    fn parse_filenames(&mut self, call_info: &CallInfo) {
+        let candidates = match &call_info.args.positional {
+            Some(values) => values
+                .iter()
+                .map(|val| val.as_string())
+                .collect::<Result<Vec<String>, _>>()
+                .unwrap_or(vec![]),
+            None => vec![],
+        };
+
+        for candidate in candidates {
+            self.add_filename(candidate);
+        }
+    }
+
+    fn parse_application(&mut self, call_info: &CallInfo) {
+        self.application = if let Some(app) = call_info.args.get("application") {
+            match app.as_string() {
+                Ok(name) => Some(name),
+                Err(_) => None,
+            }
+        } else {
+            None
+        };
+    }
+
+    pub fn parse_value(&mut self, input: &Value) {
+        if let Ok(filename) = input.as_string() {
+            self.add_filename(filename);
+        } else {
+            print_warning(format!("Could not convert '{:?}' to string", input));
+        }
+    }
+
+    #[cfg(target_os = "macos")]
+    pub fn exec(&mut self) -> Result<(), StartError> {
+        let mut args = vec![];
+        args.append(&mut self.filenames);
+
+        if let Some(app_name) = &self.application {
+            args.append(&mut vec![String::from("-a"), app_name.to_string()]);
+        }
+        exec_cmd("open", &args)
+    }
+
+    #[cfg(target_os = "windows")]
+    pub fn exec(&mut self) -> Result<(), StartError> {
+        if let Some(app_name) = &self.application {
+            for file in &self.filenames {
+                match open::with(file, app_name) {
+                    Ok(_) => continue,
+                    Err(_) => {
+                        return Err(StartError::new(
+                            "Failed to open file with specified application",
+                        ))
+                    }
+                }
+            }
+        } else {
+            for file in &self.filenames {
+                match open::that(file) {
+                    Ok(_) => continue,
+                    Err(_) => {
+                        return Err(StartError::new(
+                            "Failed to open file with default application",
+                        ))
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    #[cfg(not(any(target_os = "windows", target_os = "macos")))]
+    pub fn exec(&mut self) -> Result<(), StartError> {
+        let mut args = vec![];
+        args.append(&mut self.filenames);
+
+        if let Some(app_name) = &self.application {
+            exec_cmd(&app_name, &args)
+        } else {
+            for cmd in &["xdg-open", "gnome-open", "kde-open", "wslview"] {
```

Comment:
Looking a bit at the `open` crate, I think it will do these steps for you, depending on the OS you're on.

Were you running into any issues with it that made you need to do this step?

Reply:
Yes, the `open` crate does it, however it does not allow passing multiple filenames at once. 

Instead of having it do multiple calls, I pass all the filenames at once. 

The support hasn't been added yet, but in the future, we can pass in extra args (like say --incognito) using this model. The `open` crate doesn't support that.

Reply:
Ahh, that makes sense.